service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -0,0 +1,249 @@
+     * @return The application/json of the digital twin.
+     */
+    public <T> Observable<T> getDigitalTwin (String digitalTwinId, Class<T> clazz)
+    {
+        return getDigitalTwinWithResponse(digitalTwinId, clazz)
+                .map(response -> response.body());
+    }
+
+    /**
+     * Gets a digital twin.
+     * @param digitalTwinId The Id of the digital twin.
+     * @param clazz The class to deserialize the application/json into.
+     * @param <T> The generic type to deserialize the application/json into.
+     * @return A {@link ServiceResponseWithHeaders} representing deserialized application/json of the digital twin with {@link DigitalTwinGetHeaders}.
+     */
+    public <T> Observable<ServiceResponseWithHeaders<T, DigitalTwinGetHeaders>> getDigitalTwinWithResponse (String digitalTwinId, Class<T> clazz)
+    {
+        if(clazz == null)
+        {
+            throw new IllegalArgumentException("Parameter clazz is required and cannot be null.");
+        }
+
+        return digitalTwin.getDigitalTwinWithServiceResponseAsync(digitalTwinId)
+                .flatMap(FUNC_TO_DIGITAL_TWIN_GET_RESPONSE)
+                .flatMap(response -> {
+                    try {
+                        T genericResponse = DeserializationHelpers.castObject(objectMapper, response.body(), clazz);
+                        return Observable.just(new ServiceResponseWithHeaders<>(genericResponse, response.headers(), response.response()));
+                    } catch (JsonProcessingException e) {
+                        return Observable.error(new IotHubException("Failed to parse the resonse"));
+                    }
+
+                })
+                .subscribeOn(Schedulers.io());
+    }
+
+    /**
+     * Updates a digital twin.
+     * @param digitalTwinId The Id of the digital twin.
+     * @param digitalTwinUpdateOperations The JSON patch to apply to the specified digital twin. This argument can be created using {@link UpdateOperationUtility}.
+     * @return void.
+     */
+    public Observable<Void> updateDigitalTwin (String digitalTwinId, List<Object> digitalTwinUpdateOperations)
+    {
+        return updateDigitalTwinWithResponse(digitalTwinId, digitalTwinUpdateOperations, null)
+                .map(response -> response.body());
+    }
+
+    /**
+     * Updates a digital twin.
+     * @param digitalTwinId The Id of the digital twin.
+     * @param digitalTwinUpdateOperations The JSON patch to apply to the specified digital twin. This argument can be created using {@link UpdateOperationUtility}.
+     * @return A {@link ServiceResponseWithHeaders} with {@link DigitalTwinUpdateHeaders}.
+     */
+    public Observable<ServiceResponseWithHeaders<Void, DigitalTwinUpdateHeaders>> updateDigitalTwinWithResponse (String digitalTwinId, List<Object> digitalTwinUpdateOperations)
+    {
+        return updateDigitalTwinWithResponse(digitalTwinId, digitalTwinUpdateOperations, null);
```

Comment:
I think instead of null you should create the options with default values and pass it in.

Reply:
consider that for all APIs